spec/models/dataset_spec.rb

```diff
@@ -148,6 +148,36 @@
     expect{ Dataset.find(dataset.id) }.to raise_error(ActiveRecord::RecordNotFound)
   end
 
+  context "tests the robustness of the url evaluate response code method" do
+    {
+      'www.deadurl.com': false,
+      'http://www.deadurl.com': false,
+      'www.deadurl.com/example.csv': false,
+      'http://www.deadurl.com/example.csv': true
+    }.each_pair do |url, result|
+
+      it "should parse #{url} correctly in eval_response_code" do
+        url = url.to_s # because hash registers as symbols
```

Comment:
That's my shonky syntax, sorry. `'www.deadurl.com' => false` would give you string keys instead.